python/pyarrow/tests/test_table.py

```diff
@@ -1219,6 +1219,295 @@ def test_recordbatch_to_tensor_unsupported():
         batch.to_tensor()
 
 
+@pytest.mark.parametrize('typ', [
```

Comment:
We are adding a lot of code to this test file, while in practice it's not really testing anything new. I am wondering if we could either parametrize some existing tests for RecordBatch to run with both RecordBatch and Table (we have some other tests in this file that are parametrized that way, see eg `test_table_basics`), or otherwise only test some unique aspects of Table.to_tensor (the fact that the method exists and works, a case with multiple chunks), but for the other aspects (type promotion, null handling, etc) rely on the record batch tests.

Reply:
Added parametrization with `RecordBatch` and `Table` for mixed type test, others I left to rely on `RecordBatch` only. Then I kept one test for table with multiple chunks: https://github.com/apache/arrow/pull/41870/commits/feab6e1728ccb310c3f57374f501cbd7a4f7424d